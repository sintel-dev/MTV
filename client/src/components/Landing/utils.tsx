export const colorSchemes = {
  tag: [
    '#FFCD00', // investigate
    '#7CA8FF', // do not investigate
    '#A042FF', // postpone
    '#F64242', // problem
    '#F5FF00', // previously seen
    '#45F642', // normal
    '#C7C7C7', // untagged
  ],
};

// @TODO - later investigation for refactoring (fromIDToTag/fromTagToId)
export const tagSeq = [
  'Investigate',
  'Do not Investigate',
  'Postpone',
  'Problem',
  'Previously seen',
  'Normal',
  'Untagged',
];
const tagClassNames = ['investigate', 'not_investigate', 'postpone', 'problem', 'seen', 'normal', 'untagged'];

export const fromTagToClassName = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return tagClassNames[0];
    case tagSeq[1]:
      return tagClassNames[1];
    case tagSeq[2]:
      return tagClassNames[2];
    case tagSeq[3]:
      return tagClassNames[3];
    case tagSeq[4]:
      return tagClassNames[4];
    case tagSeq[5]:
      return tagClassNames[5];
    default:
      return 'untagged';
  }
};

export const fromIDtoTag = (id: string): string => {
  switch (id) {
    case '1':
      return tagSeq[0];
    case '2':
      return tagSeq[1];
    case '3':
      return tagSeq[2];
    case '4':
      return tagSeq[3];
    case '5':
      return tagSeq[4];
    case '6':
      return tagSeq[5];
    default:
      return 'untagged';
  }
};

export const fromTagToID = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return '1';
    case tagSeq[1]:
      return '2';
    case tagSeq[2]:
      return '3';
    case tagSeq[3]:
      return '4';
    case tagSeq[4]:
      return '5';
    case tagSeq[5]:
      return '6';
    default:
      return 'untagged';
  }
};

export const getTagColor = (tag: string): string => {
  let colorIdx: number | undefined;
  for (let i = 0; i < tagSeq.length; i += 1) {
    if (tagSeq[i] === tag) {
      colorIdx = i;
    }
  }
  if (colorIdx === undefined) {
    colorIdx = 6;
  }
  return colorSchemes.tag[colorIdx];
};
