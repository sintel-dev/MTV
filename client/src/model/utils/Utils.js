export const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
export const fromMonthToIndex = (month) => months.indexOf(month) + 1;
export const maxDaysInMonth = (currentYear, currentMonth) => new Date(currentYear, currentMonth, 0).getDate();

export function formatDate(date) {
  let hours = date.getHours();
  let minutes = date.getMinutes();
  hours %= 12;
  hours = hours || 12;
  minutes = minutes < 10 ? `0${minutes}` : minutes;
  const ampm = hours >= 12 ? 'pm' : 'am';
  let strTime = `${hours}:${minutes}`;
  const formattedTime = {
    year: date.getFullYear(),
    month: date.getMonth() + 1,
    day: date.getDate(),
    time: `${strTime}:${ampm}`,
  };
  return formattedTime;
}

export const toTimestamp = function (strDate) {
  let datum = Date.parse(strDate);
  return datum / 1000;
};

export const groupEventsByTimestamp = (events) => {
  let result = {};
  events.forEach((event) => {
    const { start_time, stop_time } = event;

    const eventStartYear = new Date(start_time * 1000).getFullYear();
    const eventStopYear = new Date(stop_time * 1000).getFullYear();
    let currentYear = eventStartYear;
    while (currentYear <= eventStopYear) {
      const yearStartDate = toTimestamp(`01/01/${currentYear} 00:00:00`);
      const yearStopDate = toTimestamp(`12/31/${currentYear} 23:59:59`);

      const eventProps = {
        id: event.id,
        start_time: start_time >= yearStartDate ? start_time : yearStartDate,
        stop_time: stop_time <= yearStopDate ? stop_time : yearStopDate,
        tag: event.tag,
        score: event.score,
      };

      if (result[currentYear]) {
        result[currentYear].events[event.id] = eventProps;
      } else {
        result[currentYear] = {
          events: { [event.id]: eventProps },
          months: {},
        };
      }
      const eventStartMonth = new Date(result[currentYear].events[event.id].start_time * 1000).getMonth() + 1;
      const eventStopMonth = new Date(result[currentYear].events[event.id].stop_time * 1000).getMonth() + 1;
      let currentMonth = eventStartMonth;

      while (currentMonth <= eventStopMonth) {
        // const maxDaysInMonth = new Date(currentYear, currentMonth, 0).getDate();
        const monthDateStart = toTimestamp(`${currentMonth}/01/${currentYear} 00:00:00`);
        const monthDateStop = toTimestamp(
          `${currentMonth}/${maxDaysInMonth(currentYear, currentMonth)}/${currentYear} 23:59:59`,
        );

        let eventPerMonth = {
          id: event.id,
          start_time: start_time >= monthDateStart ? start_time : monthDateStart,
          stop_time: stop_time <= monthDateStop ? stop_time : monthDateStop,
          tag: event.tag,
          score: event.score,
          days: {},
        };

        const eventStartDay = new Date(eventPerMonth.start_time * 1000).getDate();
        const eventStopDay = new Date(eventPerMonth.stop_time * 1000).getDate();
        let currentDay = eventStartDay;

        if (result[currentYear].months[currentMonth]) {
          result[currentYear].months[currentMonth].events[event.id] = eventPerMonth;
        } else {
          result[currentYear].months[currentMonth] = {
            events: { [event.id]: eventPerMonth },
            days: {},
          };
        }

        while (currentDay <= eventStopDay) {
          const dayDateStart = toTimestamp(`${currentMonth}/${currentDay}/${currentYear} 00:00:00`);
          const dayDateStop = toTimestamp(`${currentMonth}/${currentDay}/${currentYear} 23:59:59`);

          let day = {
            id: event.id,
            start_time: start_time >= dayDateStart ? start_time : dayDateStart,
            stop_time: stop_time <= dayDateStop ? stop_time : dayDateStop,
            tag: event.tag,
            score: event.score,
          };

          result[currentYear].months[currentMonth].days[currentDay] = day;
          currentDay += 1;
        }

        currentMonth += 1;
      }
      currentYear += 1;
    }
  });
  return result;
};

export const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
export const namePattern = /^[a-zA-Z0-9._-]+$/;

export const isEmail = (value) => emailPattern.test(value);
export const isName = (value) => namePattern.test(value);
