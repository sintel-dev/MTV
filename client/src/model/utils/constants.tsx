export const API_URL = 'http://127.0.0.1:3000/api/v1/';
export const GOOGLE_CLIEN_ID = '796048243680-341g672umjji4m4okfgppm2ogfkrh8mj.apps.googleusercontent.com';
export const SESSION_TOKEN = 'AUTHORIZATION';
