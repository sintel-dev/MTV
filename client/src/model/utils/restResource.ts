import { AxiosRequestConfig, AxiosInstance } from 'axios';

export default class Resource<T, R> {
  /**
   *
   * @param server AxiosInstance
   * @param url Recourse end point
   */
  constructor(private server: AxiosInstance, private url: string) {}

  /**
   * Get all items
   * @param data body data - in most cases, this should be {}
   * @param params URL parameters
   */
  public all(data = {}, params = {}): Promise<R> {
    const promise = this.server.get<R>(this.url, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Get one item
   * @param id the item ID (name) to be retrieved
   * @param data body data â€“ in most cases, this should be {}
   * @param params URL parameters
   */
  public find(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}`;
    const promise = this.server.get<T>(newUrl, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Create an item.
   * @param data the item data to create
   * @param params URL parameters
   */
  public create(data = {}, params = {}): Promise<T> {
    const promise = this.server.post<T>(this.url, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Delete an item.
   * @param id the item ID (name) to delete
   * @returns Promise of the response's status
   */
  public delete(id): Promise<number> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.delete<number>(newUrl);
    return promise
      .then((res) => res.status)
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Update an item.
   * @param id the item ID (name) to update
   * @param data the updated data of this item
   * @param params URL parameters
   */
  public update(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.put<T>(newUrl, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Allow customize your request.
   * @param config request config
   * @param id item id
   */
  public request<Z>(config: AxiosRequestConfig = {}, id?: string): Promise<Z> {
    let newUrl = this.url;
    if (id !== undefined) {
      newUrl += `${id}/`;
    }
    const promise = this.server.get<Z>(newUrl, config);
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }
}
