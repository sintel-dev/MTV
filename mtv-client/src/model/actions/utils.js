const baseUrl = '/api/v1/';

export const api = {
    get(url, dataType = 'json') {
        const apiUrl = `${baseUrl}${url}`;
        const promise = fetch(apiUrl, {
            mode: 'cors',
            headers: {
                'Access-Control-Allow-Origin': '*',
            },
        });
        return promise.then(response => {
            if (response.status !== 204) {
                return dataType !== 'json' ? response.text : response.json();
            }
            return null;
        })
        .catch(err => console.error(err));
    },

    // post(url, data) {
    //     const apiUrl = `${baseUrl}${url}`;
    //     return fetch(apiUrl, {
    //         mode: 'cors',
    //         method: 'POST',
    //         headers: {
    //             'Access-Control-Allow-Origin': '*',
    //             'Content-Type': 'application/json',
    //             // 'Authorization': lscache.get('authorization')
    //         },
    //         body: JSON.stringify(data)
    //     });
    // }
};
