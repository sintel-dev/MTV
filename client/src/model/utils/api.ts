import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
import Cookies from 'js-cookie';
import { API_URL, SESSION_TOKEN } from './constants';
// import { SESSION_TOKEN } from '@nx-react/session';

import Resource from './restResource';
import {
  DatasetDataType,
  DatasetsResponse,
  DatarunDataType,
  DatarunsResponse,
  ExperimentDataType,
  ExperimentsResponse,
  PipelinesResponse,
  EventsResponse,
  CommentsResponse,
  DataResponse,
} from '../types/index';
import { PipelineDataType } from '../types/pipeline';
import { EventDataType } from '../types/event';
import { CommentDataType } from '../types/comment';

export class RestClient {
  private server: AxiosInstance;

  // default config for http request
  private requestConfig: AxiosRequestConfig = {
    baseURL: API_URL,
    // timeout: 2000,
  };

  public test: Resource<any, any>;

  public datasets: Resource<DatasetDataType, DatasetsResponse>;

  public dataruns: Resource<DatarunDataType, DatarunsResponse>;

  public experiments: Resource<ExperimentDataType, ExperimentsResponse>;

  public pipelines: Resource<PipelineDataType, PipelinesResponse>;

  public events: Resource<EventDataType, EventsResponse>;

  public comments: Resource<CommentDataType, CommentsResponse>;

  public data: Resource<any, DataResponse>;

  /**
   *
   * @param config AxiosRequestConfig
   */
  constructor(config: AxiosRequestConfig) {
    this.requestConfig = { ...this.requestConfig, ...config };
    this.server = axios.create(this.requestConfig);

    // add resources
    this.test = new Resource(this.server, 'test/');
    this.datasets = new Resource(this.server, 'datasets/');
    this.dataruns = new Resource(this.server, 'dataruns/');
    this.experiments = new Resource(this.server, 'experiments/');
    this.pipelines = new Resource(this.server, 'pipelines/');
    this.events = new Resource(this.server, 'events/');
    this.comments = new Resource(this.server, 'comments/');
    this.data = new Resource(this.server, 'data/');
    this.server.interceptors.request.use(this.requestInterceptor);
    this.server.interceptors.response.use(this.responseSuccessInterceptor, this.responseFailInterceptor);
  }

  public requestInterceptor(config) {
    const authHeader = Cookies.get(SESSION_TOKEN);
    if (!authHeader) {
      return config;
    }

    return {
      ...config,
      headers: {
        ...config.headers,
        Authorization: authHeader,
      },
    };
  }

  public responseSuccessInterceptor(response) {
    return response;
  }

  public responseFailInterceptor(error) {
    if (error.response.status === 401) {
      window.location.href = '/logout';
    }

    return Promise.reject(error);
  }
}

export default new RestClient({
  baseURL: 'http://127.0.0.1:3000/api/v1/',
});
