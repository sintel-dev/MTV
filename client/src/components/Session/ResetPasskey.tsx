import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import CircularProgress from '@material-ui/core/CircularProgress';
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import { onUserResetPassKey } from 'src/model/actions/users';
import { getPasswordResetStatus } from 'src/model/selectors/users';
import { Link } from 'react-router-dom';
import { isEmail } from '../../model/utils/Utils';
import Wrapper from './Wrapper';

export interface ResetPasskeyState {
  email: string;
  emailError: boolean;
}

export interface ResetPasskeyProps {
  resetPassKyeStatus: 'loading' | 'success' | 'fail' | '';
  resetPasskey: (userData) => void;
  error: {
    message: string;
  };
}

class ResetPassKey extends Component<ResetPasskeyProps, ResetPasskeyState> {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      emailError: false,
    };
    this.onEmailChangeHandler = this.onEmailChangeHandler.bind(this);
    this.onEmailBlurHandler = this.onEmailBlurHandler.bind(this);
    this.onFormSubmitHandler = this.onFormSubmitHandler.bind(this);
  }

  onEmailChangeHandler({ target }): void {
    this.setState({
      email: target.value,
    });
  }

  onEmailBlurHandler(): void {
    const { emailError, email } = this.state;
    if (!isEmail(email) && !emailError) {
      this.setState({
        emailError: true,
      });
    }
  }

  onFormSubmitHandler(): void {
    this.props.resetPasskey({
      email: this.state.email,
    });
  }

  render() {
    if (this.props.resetPassKyeStatus === 'success') {
      return (
        <Wrapper success>
          <div className="success-wrapper">
            <p>
              If this e-mail is associated to an account, you will receive an e-mail shortly at{' '}
              <strong>{this.state.email}</strong> with your new passkey.
            </p>
            <p>
              Please follow the instructions and then <Link to="/">log in</Link>.
            </p>
          </div>
        </Wrapper>
      );
    }

    const isLoading = this.props.resetPassKyeStatus === 'loading';
    return (
      <Wrapper title="Reset PASSKEY">
        <div className="field-wrapper">
          <TextField
            fullWidth
            required
            id="email"
            label="E-mail"
            type="email"
            name="new-email"
            helperText={this.state.emailError ? 'Invalid e-mail' : null}
            disabled={isLoading}
            value={this.state.email}
            error={this.state.emailError}
            onChange={this.onEmailChangeHandler}
            onBlur={this.onEmailBlurHandler}
          />
        </div>
        <div className="button-wrapper">
          <Button
            className="button"
            fullWidth
            variant="contained"
            color="primary"
            type="submit"
            onClick={this.onFormSubmitHandler}
          >
            {isLoading && <CircularProgress color="secondary" size={28} />}
            <span>Reset Passkey</span>
          </Button>
        </div>
      </Wrapper>
    );
  }
}

export default connect(
  (state) => ({
    resetPassKyeStatus: getPasswordResetStatus(state),
  }),
  (dispatch: Function) => ({
    resetPasskey: (userEmail) => dispatch(onUserResetPassKey(userEmail)),
  }),
)(ResetPassKey);
