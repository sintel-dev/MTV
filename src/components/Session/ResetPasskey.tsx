import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import CircularProgress from '@material-ui/core/CircularProgress';
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import { onUserResetPassKey } from 'src/model/actions/users';
import { getPasswordResetStatus } from 'src/model/selectors/users';
import { Link } from 'react-router-dom';
import { RootState } from 'src/model/types';
import { isEmail } from '../../model/utils/Utils';
import Wrapper from './Wrapper';

interface ResetPasskeyState {
  email: string;
  emailError: boolean;
}

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

// @TODO - check to see if it's needed, not in use at the moment
class ResetPassKey extends Component<Props, ResetPasskeyState> {
  constructor(props: Props) {
    super(props);
    this.state = {
      email: '',
      emailError: false,
    };
    this.onEmailChangeHandler = this.onEmailChangeHandler.bind(this);
    this.onEmailBlurHandler = this.onEmailBlurHandler.bind(this);
    this.onFormSubmitHandler = this.onFormSubmitHandler.bind(this);
  }

  onEmailChangeHandler({ target }): void {
    this.setState({
      email: target.value,
    });
  }

  onEmailBlurHandler(): void {
    const { emailError, email } = this.state;
    if (!isEmail(email) && !emailError) {
      this.setState({
        emailError: true,
      });
    }
  }

  onFormSubmitHandler(): void {
    this.props.resetPasskey({
      email: this.state.email,
    });
  }

  render() {
    if (this.props.resetPassKyeStatus === 'success') {
      return (
        <Wrapper success>
          <div className="success-wrapper">
            <p>
              If this e-mail is associated to an account, you will receive an e-mail shortly at{' '}
              <strong>{this.state.email}</strong> with your new passkey.
            </p>
            <p>
              Please follow the instructions and then <Link to="/">log in</Link>.
            </p>
          </div>
        </Wrapper>
      );
    }

    const isLoading = this.props.resetPassKyeStatus === 'loading';
    return (
      <Wrapper title="Reset PASSKEY">
        <div className="field-wrapper">
          <TextField
            fullWidth
            required
            id="email"
            label="E-mail"
            type="email"
            name="new-email"
            helperText={this.state.emailError ? 'Invalid e-mail' : null}
            disabled={isLoading}
            value={this.state.email}
            error={this.state.emailError}
            onChange={this.onEmailChangeHandler}
            onBlur={this.onEmailBlurHandler}
          />
        </div>
        <div className="button-wrapper">
          <Button
            className="button"
            fullWidth
            variant="contained"
            color="primary"
            type="submit"
            onClick={this.onFormSubmitHandler}
          >
            {isLoading && <CircularProgress color="secondary" size={28} />}
            <span>Reset Passkey</span>
          </Button>
        </div>
      </Wrapper>
    );
  }
}

const mapState = (state: RootState) => ({
  resetPassKyeStatus: getPasswordResetStatus(state),
});

const mapDispatch = (dispatch: Function) => ({
  resetPasskey: (userEmail) => dispatch(onUserResetPassKey(userEmail)),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(ResetPassKey);
