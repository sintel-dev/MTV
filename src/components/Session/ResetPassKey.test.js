import React from 'react';
import { renderWithStore, TestWrapper } from 'src/tests/utils';
import ResetPasskey from './ResetPasskey';

describe('Testing Reset password component', () => {
  const resetPassword = renderWithStore(
    {},
    <TestWrapper>
      <ResetPasskey />
    </TestWrapper>,
  );
  it('Should render without crashing', () => {
    expect(resetPassword).toMatchSnapshot();
  });
});
