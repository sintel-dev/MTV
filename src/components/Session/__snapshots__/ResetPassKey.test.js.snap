// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Reset password component Should render without crashing 1`] = `
<div
  class="session-wrapper"
>
  <div
    class="session-wrapper__body"
  >
    <div
      class="session-wrapper__logo"
    >
      <svg
        height="110"
        version="1.1"
        viewBox="0 0 117 60"
        width="110"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          id="Dark-Mode"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#000000"
            fill-rule="nonzero"
            id="Register"
            transform="translate(-151.000000, -70.000000)"
          >
            <g
              id="MIT_logo"
              transform="translate(151.000000, 70.000000)"
            >
              <g
                transform="translate(7.000000, 0.000000)"
              >
                <path
                  d="M0.18,0 L0.18,59.76 M20.7,0 L20.7,40.68 M41.22,0 L41.22,59.76 M61.74,0 L61.74,11.88 M82.62,19.08 L82.62,59.76"
                  id="Shape"
                  stroke="#A31F34"
                  stroke-width="12.6"
                />
                <line
                  id="Path"
                  stroke="#8A8B8C"
                  stroke-width="12.6"
                  x1="61.74"
                  x2="61.74"
                  y1="19.08"
                  y2="59.76"
                />
                <line
                  id="Path"
                  stroke="#A31F34"
                  stroke-width="11.88"
                  x1="76.32"
                  x2="109.44"
                  y1="5.94"
                  y2="5.94"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
    <div
      class="session-wrapper__form "
    >
      <div
        class="session-wrapper__header"
      >
        <div
          class="session-wrapper__title"
        >
          Reset PASSKEY
        </div>
      </div>
      <div
        class="field-wrapper"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
            for="email"
            id="email-label"
          >
            E-mail
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
              â€‰*
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="email"
              name="new-email"
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="button-wrapper"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            <span>
              Reset Passkey
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="session-wrapper__splash"
  />
</div>
`;
